packages/adapter-cloudflare/index.js

```diff
@@ -108,28 +113,43 @@ export default function (options = {}) {
 			});
 
 			// _headers
-			if (existsSync('_headers')) {
-				copyFileSync('_headers', `${dest}/_headers`);
+			const headers_src = '_headers';
+			const headers_dest = `${dest}/_headers`;
+			if (existsSync(headers_src)) {
+				copyFileSync(headers_src, headers_dest);
 			}
-			writeFileSync(`${dest}/_headers`, generate_headers(builder.getAppPath()), { flag: 'a' });
+			writeFileSync(headers_dest, generate_headers(builder.getAppPath()), { flag: 'a' });
 
 			// _redirects
-			if (existsSync('_redirects')) {
-				copyFileSync('_redirects', `${dest}/_redirects`);
+			const redirects_src = '_redirects';
+			const redirects_dest = `${dest}/_redirects`;
+			if (existsSync(redirects_src)) {
+				copyFileSync(redirects_src, redirects_dest);
 			}
 			if (builder.prerendered.redirects.size > 0) {
-				writeFileSync(`${dest}/_redirects`, generate_redirects(builder.prerendered.redirects), {
+				writeFileSync(redirects_dest, generate_redirects(builder.prerendered.redirects), {
 					flag: 'a'
 				});
 			}
 
-			writeFileSync(`${dest}/.assetsignore`, generate_assetsignore(), { flag: 'a' });
-
 			if (building_for_cloudflare_pages) {
+				// _routes.json
+				/** @type {string[]} */
+				let redirects = [];
+				if (existsSync(redirects_dest)) {
+					const redirect_rules = readFileSync(redirects_dest, 'utf8');
```

Comment:
what's this doing? It looks like it's reading redirects created during a previous build and reusing them? why would we want to do that?

Reply:
This part is reading the `_redirects` file we've created during the current build from combining the user's redirects and our generated ones. We need to read them to add them to the `_routes.json` file `exclude` array to prevent the worker from taking over.

I got a bit confused myself when I read this. I think I should add a comment explaining these bits now that I remember.